internal/addr.go

```diff
@@ -44,8 +43,21 @@ func ParseAddr(addr string) (string, int, error) {
 	if host == "" || strings.TrimSpace(host) == "" {
 		host = defaultHost
 	}
-	if portInt < 0 { // port number should be more than 0
+	if portInt < 0 {
+		// the port number must be positive value
 		return "", 0, fmt.Errorf("invalid port number: '%d'", portInt)
 	}
```

Comment:
How about:
```
	if portInt < 0 {
		return "", 0, hzerrors.ErrInvalidAddress
	}
	if host == "" || strings.TrimSpace(host) == "" {
		host = defaultHost
	}
```

Reply:
I wouldn't support to return `hzerrors.ErrInvalidAddress` as it is not descriptive.